test/test-fs.c

```diff
@@ -642,9 +690,9 @@ TEST_IMPL(fs_file_loop) {
   /*
    * Windows XP and Server 2003 don't support symlinks; we'll get UV_ENOTSUP.
    * Starting with vista they are supported, but only when elevated, otherwise
-   * we'll see UV_EPERM.
+   * we'll see UV_EPERM. If user-mode symlinks are avaiable, check the result.
    */
-  if (r == UV_ENOTSUP || r == UV_EPERM)
+  if (usermode_symlink_available() == 0 && (r == UV_ENOTSUP || r == UV_EPERM))
```

Comment:
Again, why don't we just let the OS determine whether this call should fail, rather than duplicating the OS policy in libuv?